frontend/src/utils/VonageVideoClient/vonageVideoClient.ts

```diff
@@ -0,0 +1,207 @@
+import {
+  initSession,
+  OTError,
+  Publisher,
+  Session,
+  Stream,
+  SubscriberProperties,
+} from '@vonage/client-sdk-video';
+import { EventEmitter } from 'events';
+import {
+  Credential,
+  StreamCreatedEvent,
+  VideoElementCreatedEvent,
+  SubscriberWrapper,
+  SignalEvent,
+  SignalType,
+  SubscriberAudioLevelUpdatedEvent,
+} from '../../types/session';
+import logOnConnect from '../logOnConnect';
+import createMovingAvgAudioLevelTracker from '../movingAverageAudioLevelTracker';
+
+type VonageVideoClientEvents = {
+  archiveStarted: [string];
+  archiveStopped: [];
+  screenshareStreamCreated: [];
+  sessionDisconnected: [];
+  sessionReconnected: [];
+  sessionReconnecting: [];
+  signal: [SignalEvent];
+  ['signal:chat']: [SignalEvent];
+  ['signal:emoji']: [SignalEvent];
+  streamPropertyChanged: [];
+  subscriberVideoElementCreated: [SubscriberWrapper];
+  subscriberDestroyed: [string];
+  subscriberAudioLevelUpdated: [SubscriberAudioLevelUpdatedEvent];
+};
+
+class VonageVideoClient extends EventEmitter<VonageVideoClientEvents> {
```

Comment:
Some doc here explaining what this is and why it's here (help separate react stuff away from opentok stuff a bit)